core/platform/image/gae_image_services.py

```diff
@@ -67,7 +65,10 @@ def compress_image(image_content, scaling_factor):
         str. Returns the content of the compressed image.
     """
     if not constants.DEV_MODE:
-        height, width = get_image_dimensions(image_content)
+        image = Image.open(io.BytesIO(image_content))
+
+        image_format = image.format
+        width, height = image.width, image.height
```

Comment:
Any reason not to use the existing get_image_dimensions()?

Reply:
I will need to use image in the rest of the function so using get_image_dimension reads the image twice which I think is unnecessary reuse of code since reading bytes is slow.

Reply:
You could perhaps refactor slightly so that get_image_dimension() takes in an already-read image?

Reply:
So the way the app deals with images is encoding since when retrieved from a form it's in a string format. As far as I can tell, usage in this way is much more useful since or else we might have to refactor every usage of get_image_dimensions to first create a Pillow Image object and then convert.